libethereum/TransactionQueue.cpp

```diff
@@ -57,34 +57,16 @@ TransactionQueue::~TransactionQueue()
 
 ImportResult TransactionQueue::import(bytesConstRef _transactionRLP, IfDropped _ik)
 {
-	// Check if we already know this transaction.
-	h256 h = sha3(_transactionRLP);
-
 	Transaction t;
-	ImportResult ir;
+	try
 	{
-		UpgradableGuard l(m_lock);
-
-		ir = check_WITH_LOCK(h, _ik);
-		if (ir != ImportResult::Success)
-			return ir;
-
-		try
-		{
-			// Check validity of _transactionRLP as a transaction. To do this we just deserialise and attempt to determine the sender.
-			// If it doesn't work, the signature is bad.
-			// The transaction's nonce may yet be invalid (or, it could be "valid" but we may be missing a marginally older transaction).
-			t = Transaction(_transactionRLP, CheckTransaction::Everything);
-			UpgradeGuard ul(l);
-//			cdebug << "Importing" << t;
-			ir = manageImport_WITH_LOCK(h, t);
-		}
-		catch (...)
-		{
-			return ImportResult::Malformed;
-		}
+		Transaction(_transactionRLP, CheckTransaction::Everything);
+	}
+	catch (...)
```

Comment:
Please catch some type. Do we know what is expected from Transaction constructor? If that set is small, can we move the `return ...` here too?